sound/soc/sof/intel/hda-ipc.c

```diff
@@ -160,11 +161,10 @@ irqreturn_t hda_dsp_ipc4_irq_thread(int irq, void *context)
 			spin_unlock_irq(&sdev->ipc_lock);
 		} else {
 			/* Notification received */
-			struct sof_ipc4_msg data = {{ 0 }};
 
 			data.primary = primary;
 			data.extension = extension;
-			sdev->ipc->msg.rx_data = &data;
+			sdev->ipc->msg.rx_data = &null_data;
```

Comment:
a bit silly that cppcheck got this one wrong...

Reply:
for the record, I'm using `cppcheck` locally with these parameters and it is not complaining:
`--platform=unix64 --force --max-configs=1024 --inconclusive --enable=all --suppress=variableScope --suppress=shiftTooManyBitsSigned --suppress=arithOperationsOnVoidPointer --suppress=bitwiseOnBoolean`
Should I extend the parameters (without options cppcheck is all green for hda-ipc.c)?

Reply:
I also use the same parameters @ujfalusi 

`````
cppcheck --platform=unix64 --force --max-configs=1024 --inconclusive --enable=all --suppress=variableScope --suppress=shiftTooManyBitsSigned --suppress=arithOperationsOnVoidPointer --suppress=bitwiseOnBoolean sound/soc/sof/
````

cppcheck --version
Cppcheck 2.3

Reply:
@plbossart, interesting
```
cppcheck --version
Cppcheck 2.7.4
```
on my side.